sort.c

```diff
@@ -10,30 +10,52 @@
 
 #include "redisraft.h"
 
-static void replySortedArray(RedisModuleCtx *ctx, RedisModuleCallReply * reply)
+static void replySortedArray(RedisModuleCtx *ctx, RedisModuleCallReply * reply, int reply_type)
 {
     size_t len = RedisModule_CallReplyLength(reply);
-    if (len == 0) {
-        RedisModule_ReplyWithNullArray(ctx);
-        return;
-    }
 
-    RedisModuleDict * dict = RedisModule_CreateDict(ctx);
-    for(int i=0; i < len; i++) {
-        RedisModuleCallReply * entry = RedisModule_CallReplyArrayElement(reply, i);
+    RedisModuleDict *dict = RedisModule_CreateDict(ctx);
+    for(size_t i = 0; i < len; i++) {
+        long *val;
+        RedisModuleCallReply * entry;
+        switch (reply_type) {
+            case REDISMODULE_REPLY_ARRAY:
+                entry = RedisModule_CallReplyArrayElement(reply, i);
+                break;
+            case REDISMODULE_REPLY_SET:
+                entry = RedisModule_CallReplySetElement(reply, i);
+                break;
+        }
```

Comment:
Consider being more defensive in case an unexpected type is specified (or passed).